src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,273 @@
+#include "jit_aware_lqp_translator.hpp"
+
+#include <boost/range/combine.hpp>
+
+#include <queue>
+#include <unordered_set>
+
+#include "constant_mappings.hpp"
+#include "operators/jit_operator/operators/jit_filter.hpp"
+#include "operators/jit_operator/operators/jit_read_tuple.hpp"
+#include "operators/jit_operator/operators/jit_write_tuple.hpp"
+#include "projection_node.hpp"
+#include "storage/storage_manager.hpp"
+#include "stored_table_node.hpp"
+
+namespace opossum {
+
+std::shared_ptr<AbstractOperator> JitAwareLQPTranslator::translate_node(
+    const std::shared_ptr<AbstractLQPNode>& node) const {
+  uint32_t num_jittable_nodes{0};
+  std::unordered_set<std::shared_ptr<AbstractLQPNode>> input_nodes;
+
+  // Traverse query tree until a non-jitable nodes is found in each branch
+  _breadth_first_search(node, [&](auto& current_node) {
```

Comment:
Is the BFS aspect relevant for the result or is it an irrelevant implementation detail?

Reply:
(moving this to the discussion in the `_breadth_first_search` implementation)